src/mame/drivers/segas18.cpp

```diff
@@ -564,9 +565,10 @@ WRITE16_MEMBER( segas18_state::lghost_custom_io_w )
 
 WRITE8_MEMBER( segas18_state::lghost_gun_recoil_w )
 {
-	output().set_value("P1_Gun_Recoil", (~data & 0x01));
-	output().set_value("P2_Gun_Recoil", (~data & 0x02)>>1);
-	output().set_value("P3_Gun_Recoil", (~data & 0x04)>>2);
+	for (int i = 0; i < 3; i++)
+	{
+		m_gun_recoil[i] = (~data & (1<<i))>>i;
```

Comment:
This would be easier to read if you just did `BIT(~data, i)` instead of the explicit shifts.